### PR TITLE
#56: Support column references in value expressions

### LINKED ISSUE #56: Support column references in value expressions
## Situation

Column references &mdash; i.e. names of columns in an expression &mdash; are valid value expressions. That means that a column reference can stand where a value expression is expected.

Example:

```sql
MERGE INTO ... AS ...
USING ... AS "SRC" ON ...
WHEN NOT MATCHED THEN INSERT ("ID", ...) VALUES ("SRC"."ID", ...)
```

In the example above the content of `VALUES` is a list of value expressions. Column references are useful in that scenario.

## Acceptance Criteria

1. Column references are accepted where value expressions are required.

### PR DESCRIPTION
**Pull request by redcatbear**
Closes #56.

### CODE REVIEW

File: src/main/java/com/exasol/sql/dml/insert/AbstractInsertValueTable.java [hunk 1]
```diff
@@ -82,6 +83,19 @@ public T values(final int... values) {
         return self();
     }
 
+    /**
+     * Insert a list of value expressions.
+     *
+     * @param expressions integer values to be inserted
```
Comment (morazow) [inline src/main/java/com/exasol/sql/dml/insert/AbstractInsertValueTable.java]: Change `@param expressions integer values ...` to `@param expressions expression values ...`.

File: doc/user_guide/statements/merge.md [hunk 2]
```diff
@@ -68,6 +68,10 @@ merge.using("src") //
     .where(gt(column("src", "c5"), integerLiteral(1000)));
 ```
 
+### Using Value Tables in `MERGE` Insert Clauses
+
+in the insert clause of a merge statement you can use `VALUES` to insert [value tables](common_constructs/value_tables.md).
```
Comment (morazow) [inline doc/user_guide/statements/merge.md]:
```suggestion
In the insert clause of a merge statement, you can use `VALUES` to insert [value tables](common_constructs/value_tables.md).
```

Comment (morazow) [inline doc/user_guide/statements/merge.md]:
```suggestion
in the insert clause of a merge statement you can use `VALUES` to insert [value tables](../common_constructs/value_tables.md).
```

File: doc/user_guide/common_constructs/value_tables.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# Value Tables
+
+Value tables are lists of content definitions for rows. They can appear in [`INSERT`](../statements/insert.md) and [`MERGE`](../statements/merge.md) statements.
```
Comment (morazow) [inline doc/user_guide/common_constructs/value_tables.md]: `insert.md` document is missing.

Comment (redcatbear) [reply doc/user_guide/common_constructs/value_tables.md]: Created issue #59.

File: doc/user_guide/statements/merge.md (hunk 2 above)
Comment (redcatbear) [reply doc/user_guide/statements/merge.md]: Fixed.

File: src/main/java/com/exasol/sql/dml/insert/AbstractInsertValueTable.java (hunk 1 above)
Comment (redcatbear) [reply src/main/java/com/exasol/sql/dml/insert/AbstractInsertValueTable.java]: Fixed.

**Pull request merged**